pkg/kubelet/kubelet.go

```diff
@@ -1597,9 +1597,15 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 			}
 		}
 		if mirrorPod == nil || deleted {
-			glog.V(3).Infof("Creating a mirror pod for static pod %q", format.Pod(pod))
-			if err := kl.podManager.CreateMirrorPod(pod); err != nil {
-				glog.Errorf("Failed creating a mirror pod for %q: %v", format.Pod(pod), err)
+			node, err := kl.GetNode()
+			if err != nil {
+				glog.V(4).Infof("No need to create a mirror pod, since node %q has been removed from the cluster", kl.nodeName)
+			}
+			if err == nil && node.DeletionTimestamp == nil {
```

Comment:
make this an `else if` rather than checking `err` twice.

Reply:
Actually, I would move the deletion check to the above if, and just make this the else condition. I.e.
```go
if err != nil || node.DeletionTimestamp != nil {
...
} else {
...
}
```